static-assets/components/cstudio-forms/controls/rte.js

```diff
@@ -376,70 +329,46 @@ CStudioAuthoring.Module.requireModule(
             min_height: _thisControl.rteHeight,
             theme: 'silver',
             plugins: pluginList,
-            toolbar1: toolbarConfig1,
-            toolbar2: toolbarConfig2,
-            toolbar3: toolbarConfig3,
-            toolbar4: toolbarConfig4,
             toolbar_sticky: true,
             image_advtab: true,
             encoding: 'xml',
             relative_urls: false,
             remove_script_host: false,
             convert_urls: false,
-            readonly: _thisControl.readonly,
-            force_br_newlines: forceBRTags,
-            forced_root_block: forceRootBlockPTag,
+            readonly: _thisControl.readonly, // comes from control props (not xml config)
+            force_br_newlines: forceBRTags, // comes from control props (not xml config)
+            forced_root_block: forceRootBlockPTag, // comes from control props (not xml config)
             remove_trailing_brs: false,
             media_live_embeds: true,
             autoresize_on_init: false,
             autoresize_bottom_margin: 0,
-            extended_valid_elements: extendedValidElements,
-            browser_spellcheck: this.enableSpellCheck,
-            contextmenu: !this.enableSpellCheck,
-            valid_children: validChildren,
-            image_uploadtab: this.editorImageDatasources.length > 0,
-            craftercms_paste_cleanup: rteConfig.craftercmsPasteCleanup?.trim() !== 'false',
-
-            menu: {
-              tools: { title: 'Tools', items: 'tinymcespellchecker code acecode wordcount' }
-            },
-
+            contextmenu: !this.enableSpellCheck, // comes from control props (not xml config)
+            image_uploadtab: this.editorImageDatasources.length > 0, // comes from control props (not xml config)
+            craftercms_paste_cleanup: rteConfig?.tinymceOptions?.craftercmsPasteCleanup ?? true, // If doesn't exist or if true => true
```

Comment:
This should always be a boolean. Is the `craftercmsPasteCleanup` pre-parsed to boolean?
Also, I don't think we're applied this changes to XB's rte. Please add the `craftercms_paste_extension` and this config.

Reply:
Yeah those values are boolean.
I'll add it to XB's rte.